### PR TITLE
Add more potential matches for regex of input files

### PR DESCRIPTION
**Pull request by GiudGiud**
Attempts a bug fix for #17426 #19137

doc doesnt build locally (someone teach me maybe), so going to debug with this PR

### CODE REVIEW

**Comment by GiudGiud** [comment]
@cticenhour this should be ready for review.
As the new log shows, quite a few input file objects are not matched to a class, and therefore that class' documentation does not list these inputs. I think it's just a matter of being more consistent between the C++ object names and the input file names. Might require some renaming to fix. Not the scope for this PR imo

**Comment by GiudGiud** [comment]
umm only apps failures at this point
https://civet.inl.gov/job/1042583/

I think I ll need to do some patches to make it work, possibly break next

**Comment by GiudGiud** [comment]
@cticenhour this is ready!
so happy rn haha

**Comment by cticenhour** [comment]
I've been seeing the intense force pushes - good job! I'll give a formal review this afternoon 😄

**Comment by GiudGiud** [comment]
seems I broke it haha
I ll look more. we want a lot of things from this

**Comment by GiudGiud** [comment]
ok should be good now

File: python/MooseDocs/common/build_class_database.py [hunk 1]
```diff
@@ -106,6 +123,15 @@ def _match_child(objects, filename, match):
 def _match_input(objects, filename, match):
     """Input use match function."""
     key = match.group('key')
-    if key in objects:
-        filename = os.path.relpath(filename, MooseDocs.ROOT_DIR)
-        objects[key].inputs.add(filename)
+
+    # AD instances are caught for both the regular and AD object. This is done because:
+    # - in all cases I know of, the AD/nonAD are from the same template
```
[the remark this answers is not in the document]
Comment (cticenhour) [reply python/MooseDocs/common/build_class_database.py:128]: Could we improve this logic by looking for and flagging dict objects post `_match_definition` that have the same matched header?

[the remark this answers is not in the document]
Comment (cticenhour) [reply python/MooseDocs/common/build_class_database.py:128]: Might be beyond the scope of this PR as you do successfully address the raised issues, and I cannot think of a case where there are AD/nonAD objects, templated or not, that have differing parameters and perform the same function.

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: python/MooseDocs/common/build_class_database.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GiudGiud) [reply python/MooseDocs/common/build_class_database.py:128]: sure can do, pushing a commit now

**Review by cticenhour — APPROVED** [review]
Comment was addressed, and new class database now functions as expected for both separate and templated AD/nonAD pairs. Thanks for doing this!

**Pull request merged**